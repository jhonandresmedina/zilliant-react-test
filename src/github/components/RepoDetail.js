//@vendor
import React from 'react';
import { Paper, Button, DataTable, TableRow, TableColumn, TableBody } from 'react-md';
import { Map } from 'immutable';

const RepoDetail = ({ repo, unselectRepo }) => {
    const children = [];
    repo.forEach((value, key) => {
        const element =
            typeof value !== 'object' && !Map.isMap(value) ? (
                <TableRow key={key}>
                    <TableColumn>{key.toUpperCase()}</TableColumn>
                    <TableColumn>
                        {key.endsWith('url') ? (
                            <a href={value}>{value}</a>
                        ) : value !== null ? (
                            value.toString()
                        ) : (
                            'None'
                        )}
                    </TableColumn>
                </TableRow>
            ) : null;
        children.push(element);
    });

    return (
        <div style={{ paddingTop: '20px', paddingRight: '20px', width: '100%' }}>
            <Button onClick={unselectRepo} style={{ marginBottom: '20px' }} mini floating>
                arrow_back
            </Button>
            <Paper style={{ padding: '20px' }}>
                <h2>{repo.get('name')}</h2>
                <DataTable plain>
                    <TableBody>{children}</TableBody>
                </DataTable>
            </Paper>
        </div>
    );
};

export default RepoDetail;
