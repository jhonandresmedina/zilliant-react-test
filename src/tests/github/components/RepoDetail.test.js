//@vendor
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { fromJS } from 'immutable';

//@components
import RepoDetail from '../../../github/components/RepoDetail';

Enzyme.configure({ adapter: new Adapter() });

const buildComponent = () => {
    let repo = fromJS({
        id: 77777,
        name: 'repo name',
        full_name: 'reactjs app',
        language: 'ES',
    });

    const unselectRepo = jest.fn();

    return <RepoDetail repo={repo} unselectRepo={unselectRepo} />;
};

describe('RepoDetail test', () => {
    it('renders correctly', () => {
        const wrapper = shallow(buildComponent());
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
