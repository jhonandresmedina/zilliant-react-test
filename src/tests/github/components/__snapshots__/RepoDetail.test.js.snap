// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoDetail test renders correctly 1`] = `
<div
  style={
    Object {
      "paddingRight": "20px",
      "paddingTop": "20px",
      "width": "100%",
    }
  }
>
  <withInk(withTooltip(Button))
    floating={true}
    inkTransitionEnterTimeout={450}
    inkTransitionLeaveTimeout={300}
    inkTransitionOverlap={150}
    mini={true}
    onClick={[MockFunction]}
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  >
    arrow_back
  </withInk(withTooltip(Button))>
  <Paper
    component="div"
    style={
      Object {
        "padding": "20px",
      }
    }
    zDepth={1}
  >
    <h2>
      repo name
    </h2>
    <DataTable
      checkboxHeaderLabel="Toggle All Rows"
      checkboxLabelTemplate="Toggle row {{row}}"
      defaultSelectedRows={Array []}
      fixedDividers={true}
      fixedFooter={false}
      fixedHeader={false}
      footerHeight={48}
      fullWidth={true}
      headerHeight={56}
      indeterminateIcon={
        <FontIcon
          iconClassName="material-icons"
        >
          indeterminate_check_box
        </FontIcon>
      }
      plain={true}
      responsive={true}
      selectableRows={true}
    >
      <TableBody>
        <TableRow
          key="id"
        >
          <withTooltip(TableColumn)>
            ID
          </withTooltip(TableColumn)>
          <withTooltip(TableColumn)>
            77777
          </withTooltip(TableColumn)>
        </TableRow>
        <TableRow
          key="name"
        >
          <withTooltip(TableColumn)>
            NAME
          </withTooltip(TableColumn)>
          <withTooltip(TableColumn)>
            repo name
          </withTooltip(TableColumn)>
        </TableRow>
        <TableRow
          key="full_name"
        >
          <withTooltip(TableColumn)>
            FULL_NAME
          </withTooltip(TableColumn)>
          <withTooltip(TableColumn)>
            reactjs app
          </withTooltip(TableColumn)>
        </TableRow>
        <TableRow
          key="language"
        >
          <withTooltip(TableColumn)>
            LANGUAGE
          </withTooltip(TableColumn)>
          <withTooltip(TableColumn)>
            ES
          </withTooltip(TableColumn)>
        </TableRow>
      </TableBody>
    </DataTable>
  </Paper>
</div>
`;
